upgrade.py

```diff
@@ -149,41 +149,58 @@ def init_id_maps(src_root, dst_root):
                         logger.error('Failed to parse: ' + line)
                         logger.logException(e)
 
-        # Copy ownership from a path in a source root to another path in a
-        # destination root. The ownership is copied such that it is not
-        # affected by changes in the underlying uid/gid.
-        def copy_ownership(src_root, src_path, dst_root, dst_path):
+        # Copy attributes from a path in a source root to another path in a
+        # destination root. 
+        # - The ownership is copied such that it is notaffected by changes 
+        #   in the underlying uid/gid. Or it's set to the specified.
+        # - The mode is set to the specified.
```

Comment:
Not an English native speaker but I think `specified` needs something after like `specified one`. I would change to `The mode is set to the specified one if present`.

Reply:
Will fix.